src/actions/PrecursorAction.C

```diff
@@ -233,7 +244,38 @@ PrecursorAction::act()
         std::string kernel_name = "VelocityFunctionOutflowBC_" + var_name + "_" + _object_suffix;
         _problem->addBoundaryCondition("VelocityFunctionOutflowBC", kernel_name, params);
       }
+      // INFLOW
+      if (getParam<bool>("loop_precs"))
+      {
+          // this SHOULD work for both constant and nonconstant flows as long as
+          // nonconstant flows implemented in the Controls module by
+          // setting values called uu, vv, ww.
```

Comment:
is there a way to automate this? one good rule is to not repeat data entry (humans are the buggiest part of any process).

Reply:
Ah, so, these data would only be entered once, in the controls module. For constant flows, when no Controls are used, these values should be set through the Precursors block. Either way, it only appears once in the input file.

I may miss what you're saying though.

Reply:
ok, I see what you're saying.